// react-native-vkontakte-login/scripts/postlink.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var fs_1 = __importDefault(require("fs"));
var inquirer_1 = __importDefault(require("inquirer"));
var path_1 = __importDefault(require("path"));
var android_1 = require("./android");
var findPodfile_1 = __importDefault(require("./findPodfile"));
var ios_1 = require("./ios");
function loadVkAppId() {
    var envFile = path_1.default.join(process.cwd(), '.env');
    if (fs_1.default.existsSync(envFile)) {
        var envFileContent = fs_1.default.readFileSync(envFile).toString();
        var regex = /^VK_APP_ID=(\d+)/gm;
        var match = regex.exec(envFileContent);
        if (match) {
            return match[1];
        }
    }
    return;
}
function saveVkAppId(appId, replaceExisting) {
    var envFile = path_1.default.join(process.cwd(), '.env');
    var keyValue = "VK_APP_ID=" + appId;
    var envFileContent = keyValue;
    if (fs_1.default.existsSync(envFile)) {
        envFileContent = fs_1.default.readFileSync(envFile).toString();
        envFileContent = replaceExisting ?
            envFileContent.replace(/^VK_APP_ID=\d+/gm, keyValue) :
            envFileContent + "\n" + keyValue;
    }
    fs_1.default.writeFileSync(envFile, envFileContent);
}
function postlink() {
    return __awaiter(this, void 0, void 0, function () {
        var vkAppId, podfile, answers;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    vkAppId = loadVkAppId();
                    podfile = findPodfile_1.default();
                    return [4, inquirer_1.default.prompt([
                            {
                                name: 'automate',
                                type: 'confirm',
                                message: '[react-native-vkontakte-login] Automatically modify Android and iOS projects?',
                                default: true,
                            },
                            {
                                name: 'appId',
                                type: 'input',
                                message: 'Enter your VK Application ID',
                                default: vkAppId,
                                validate: function (appId) {
                                    var valid = /\d+/.test(appId);
                                    return valid || 'VK Application id can only contain digits';
                                },
                                when: function (hash) { return hash.automate; },
                            },
                            {
                                name: 'modXcode',
                                type: 'confirm',
                                message: 'It seems that your project doesn\'t use Cocoapods. Do you want to modify XCode project file?',
                                default: true,
                                when: function (hash) { return hash.automate && podfile === null; },
                            },
                        ])];
                case 1:
                    answers = _a.sent();
                    if (answers.appId && answers.appId !== vkAppId) {
                        saveVkAppId(answers.appId, !!vkAppId);
                    }
                    if (answers.automate) {
                        android_1.postlinkAndroid();
                        ios_1.postlinkIOS(answers.appId, answers.modXcode);
                    }
                    return [2];
            }
        });
    });
}
postlink();
